### PR TITLE
Build dependency missing from aws-iot-device-sdk-cpp-v2

### PR DESCRIPTION
**Pull request by Kevin-Tacheny**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,6 +15,7 @@ endif()
 
 set(IS_SUBDIRECTORY_INCLUDE true)
 find_package(aws-crt-cpp REQUIRED)
+find_package(aws-c-cal REQUIRED)
```
Comment (JonathanHenson) [inline CMakeLists.txt]: This should be getting exported by aws-crt-cpp and not exposed outside of that. So if it’s not being picked up, we need to fix it there.

**Comment by Kevin-Tacheny** [comment]
I agree,  this really was a temp did for us to move forward.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by JonathanHenson** [comment]
now I'm actually confused. The HEAD of aws-crt-cpp already exports aws-c-cal. Were you just finding this error on the tag we recommended?

**Comment by Kevin-Tacheny** [comment]
Yes there is some sort of issue that causes aws-c-cal to not get picked up by the linker (from the v2 build).
I found the issue on the master of the v2 repository. (at least when using the visual studio cmake feature).

Best Regards,
Kevin Tacheny

**Comment by Kevin-Tacheny** [comment]
Here is the failure when building v2:  As I am new to CMake I don't know how this should be fixed and/or where.  My explicit inclusion of aws-c-cal was so we could move forward.  


cmake -DCMAKE_INSTALL_PREFIX=~/projects/out -DCMAKE_PREFIX_PATH=~/projects/out -DBUILD_SHARED_LIBS=ON -DCMAKE_BUILD_TYPE=Debug ../                      -- The CXX compiler identification is GNU 6.3.0                                                                                                         -- Check for working CXX compiler: /usr/bin/c++                                                                                                         -- Check for working CXX compiler: /usr/bin/c++ -- works                                                                                                -- Detecting CXX compiler ABI info                                                                                                                      -- Detecting CXX compiler ABI info - done                                                                                                               -- Detecting CXX compile features                                                                                                                       -- Detecting CXX compile features - done                                                                                                                -- Found LibCrypto: /usr/lib/arm-linux-gnueabihf/libcrypto.so                                                                                           -- Libcrypto Include Dir: /usr/include                                                                                                                  -- Configuring done                                                                                                                                     CMake Error at jobs/CMakeLists.txt:42 (add_library):                                                                                                      Target "IotJobs-cpp" links to target "AWS::aws-c-cal" but the target was                                                                                not found.  Perhaps a find_package() call is missing for an IMPORTED                                                                                    target, or an ALIAS target is missing?                                                                                                                                                                                                                                                                                                                                                                                                                                CMake Error at shadow/CMakeLists.txt:42 (add_library):                                                                                                    Target "IotShadow-cpp" links to target "AWS::aws-c-cal" but the target was                                                                              not found.  Perhaps a find_package() call is missing for an IMPORTED                                                                                    target, or an ALIAS target is missing?                                                                                                                                                                                                                                                                                                                                                                                                                                CMake Error at samples/jobs/describe_job_execution/CMakeLists.txt:10 (add_executable):                                                                    Target "describe-job-execution" links to target "AWS::aws-c-cal" but the                                                                                target was not found.  Perhaps a find_package() call is missing for an                                                                                  IMPORTED target, or an ALIAS target is missing?                                                                                                                                                                                                                                                                                                                                                                                                                       CMake Error at samples/shadow/shadow_sync/CMakeLists.txt:10 (add_executable):                                                                             Target "shadow-sync" links to target "AWS::aws-c-cal" but the target was                                                                                not found.  Perhaps a find_package() call is missing for an IMPORTED                                                                                    target, or an ALIAS target is missing?                                                                                                                                                                                                                                                                                                                                                                                                                                CMake Error at jobs/CMakeLists.txt:42 (add_library):                                                                                                      Target "IotJobs-cpp" links to target "AWS::aws-c-cal" but the target was                                                                                not found.  Perhaps a find_package() call is missing for an IMPORTED                                                                                    target, or an ALIAS target is missing?                                                                                                                                                                                                                                                                                                                                                                                                                                CMake Error at shadow/CMakeLists.txt:42 (add_library):                                                                                                    Target "IotShadow-cpp" links to target "AWS::aws-c-cal" but the target was                                                                              not found.  Perhaps a find_package() call is missing for an IMPORTED                                                                                    target, or an ALIAS target is missing?
--

**Comment by justinboswell** [comment]
The above PR (which is now merged) should resolve the compile issue in the samples.

**Comment by justinboswell** [comment]
The above linked PR, which is now merged to master, should resolve the issues in this PR. If not, please open an issue ticket.

**Pull request closed without merging**